core/unit_test/tools/include/ToolTestingUtilities.hpp

```diff
@@ -267,9 +267,19 @@ struct invoke_helper {
  */
 inline MatchDiagnostic check_match(event_vector::size_type events_scanned,
                                    const event_vector& events) {
-  return (events_scanned == events.size())
-             ? MatchDiagnostic{true}
-             : MatchDiagnostic{false, {"Wrong number of events encountered"}};
+// Workaround intel/17 internal error
+// See https://github.com/kokkos/kokkos/issues/4612
+#if defined(KOKKOS_COMPILER_INTEL) && (KOKKOS_COMPILER_INTEL < 1800)
+  auto result =
+      ((events_scanned == events.size())
+           ? MatchDiagnostic{true}
+           : MatchDiagnostic{false, {"Wrong number of events encountered"}});
+  return result;
+#else
+  return ((events_scanned == events.size())
+              ? MatchDiagnostic{true}
+              : MatchDiagnostic{false, {"Wrong number of events encountered"}});
+#endif
```

Comment:
Why not just
```suggestion
  auto result =
      ((events_scanned == events.size())
           ? MatchDiagnostic{true}
           : MatchDiagnostic{false, {"Wrong number of events encountered"}});
  return result;
```
?

Reply:
That's fine too :)

Reply:
I force pushed the cleaner update 08549b84299a2fc2e6041527ca4db72926d7ffd7